client_it_test.go

```diff
@@ -213,3 +214,95 @@ func TestClient_AddDistributedObjectListener(t *testing.T) {
 		mu.Unlock()
 	})
 }
+
+func TestClusterReconnection(t *testing.T) {
+	ctx := context.Background()
+	cls := it.StartNewClusterWithOptions("go-cli-test-cluster", 15701, 3)
+	mu := &sync.Mutex{}
+	events := []hz.LifecycleState{}
+	config := cls.DefaultConfig()
+	config.AddLifecycleListener(func(event hz.LifecycleStateChanged) {
+		mu.Lock()
+		events = append(events, event.State)
+		mu.Unlock()
+	})
+	c, err := hz.StartNewClientWithConfig(ctx, config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	time.Sleep(5 * time.Second)
+	cls.Shutdown()
+	time.Sleep(5 * time.Second)
+	cls = it.StartNewClusterWithOptions("go-cli-test-cluster", 15701, 3)
+	time.Sleep(5 * time.Second)
+	cls.Shutdown()
+	c.Shutdown(ctx)
+	mu.Lock()
+	defer mu.Unlock()
+	target := []hz.LifecycleState{
+		hz.LifecycleStateStarting,
+		hz.LifecycleStateClientConnected,
+		hz.LifecycleStateStarted,
+		hz.LifecycleStateClientDisconnected,
+		hz.LifecycleStateClientConnected,
+		hz.LifecycleStateClientDisconnected,
+		hz.LifecycleStateShuttingDown,
+		hz.LifecycleStateShutDown,
+	}
+	fmt.Println("target :", target)
+	fmt.Println("events :", events)
```

Comment:
Leftover or printing to stdout during tests is a common thing in our test suite? There is a similar one below

Reply:
`assert.Equal` prints the numeric values of lifecycle events, although we have nice string representations for them. I've replaces prints with logs at: https://github.com/hazelcast/hazelcast-go-client/pull/590/commits/b968570d55f526d7006a824378208c9c5fa8cf5b